mlibrary/pride
# Pride.FieldTree.Tag and Pride.FieldTree.insideFieldNodes

# Overview
`Pride.FieldTree.Tag` has been simplified, and tests have been written for `type`, `childTypes`, and `extension`.

## `extension`
The original code written for `extension` produced empty results, even if children existed.

## `Pride.FieldTree.insideFieldNodes`
The values used for `insideFieldNodes` also mirrored what's used in `Pride.FieldTree.Field`. The new property was created to be imported into all files that use the value to reduce repeated code.

## Testing
1. Build the repository (`npm run build`).
2. Run the tests to make sure they pass (`npm run test`).
   * Break the new/updated unit tests to make sure they're working properly.
3. Navigate to your local [Search repository](https://github.com/mlibrary/search) and apply the newly generated files:
   1. Open `./package.json` and add `#Pride-FieldTree-Tag` at the end of the `pride` dependency URL:
      ```bash
      "pride": "git+https://github.com/mlibrary/pride.git#Pride-FieldTree-Tag"
      ``` 
   2. Do a clean install of Search:
      ```bash
      rm -rf node_modules && rm package-lock.json && npm install
      ``` 
   3. Start Search (`npm start`) and look around [the site](http://localhost:3000/everything). Check to see if there are any console errors, and everything still works as expected.
- Make sure the PR is consistent in these browsers:
